src/ui/FieldImage/FieldImage.ts

```diff
@@ -0,0 +1,85 @@
+import { Component, Initialization } from '../../Core';
+import { exportGlobally } from '../../GlobalExports';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { IQueryResult } from '../../rest/QueryResult';
+import { FieldValue, IFieldValueOptions } from '../FieldValue/FieldValue';
+import { ComponentOptions, IFieldOption } from '../Base/ComponentOptions';
+
+export interface IFieldImage {
+  field?: IFieldOption;
+  width?: string;
+  height?: string;
+}
+
+/**
+ * The FieldImage component displays the value of a field associated to its parent search result.
+ * It is use strictly for displaying an image.
+ *
+ * This component is a result template component (see [Result Templates](https://developers.coveo.com/x/aIGfAQ)).
+ *
+ * A common use of this component is to display the image of a product for sale.
+ */
+export class FieldImage extends Component {
+  static ID = 'FieldImage';
+  static doExport = () => {
+    exportGlobally({
+      FieldImage: FieldImage
+    });
+  };
+
+  /**
+   * The options for the component
+   * @componentOptions
+   */
+  static options: IFieldImage = {
+    /**
+     * Specifies the field that the FieldImage should display.
+     *
+     * Specifying a value for this parameter is required in order for the FieldImage component to work.
+     *
+     * **Default value:** `@field`
+     */
+    field: ComponentOptions.buildFieldOption({ defaultValue: '@field', required: true }),
+    /**
+     * The width of the image
+     */
+    width: ComponentOptions.buildStringOption(),
```

Comment:
What do you think of adding defaults for the width and height? Also, should they be number options instead of strings (here I am assuming all values are in `pixels` and the user cannot put arbitrary units)?

Reply:
You can pass `pixels`, yes but you can also pass `%` so I need the string.
I think I'll add in the description that it can be either `pixels` or` percentages`

Reply:
How useful will that be?
I think every client will want to change it and setting it is super easy now so I don't think it will be super useful to add a default.

Reply:
also, if we set a default, it will destroy the original ratio of the image
I guess we could only set one of the options if necessary.

Reply:
Since two units are allowed, a string is the way to go 👍 

Good point regarding the ratio. I guess a different question I would ask is does the image display if neither the width or height options are specified?

If it does not display (because the height and width are 0?), then I believe we should have a default for at least one of the dimensions. Having a user see an image that is the wrong size is easier to debug than not seeing one at all.

On the other hand, if the image displays, but it's too big/takes a lot of space in the template, then having a reasonable default would be a nicer experience, but less important than the case where the image does not display.

Reply:
Tried it this afternoon and it displayed at its original dimension.
so it is oversized.

What I will do is give a default to the `width` of `50px`. The `height` will adjust to respect the ratio.